add-on/chrome/bootstrap.js

```diff
@@ -481,11 +481,64 @@ var WindowListener = {
         gBrowser.tabContainer.removeEventListener("TabSelect", this);
         gBrowser.removeEventListener("DOMTitleChanged", this);
         gBrowser.removeEventListener("mousemove", this);
+        this.removeRemoteCursor();
         this._listeningToTabSelect = false;
         this._browserSharePaused = false;
       },
 
       /**
+       *  If sharing is active, paints and positions the remote cursor
+       *  over the screen
+       *
+       *  @param cursorData Object with the correct position for the cursor
+       *                    {
+       *                      ratioX: position on the X axis (percentage value)
+       *                      ratioY: position on the Y axis (percentage value)
+       *                    }
+       */
+      addRemoteCursor: function(cursorData) {
+        if (!this._listeningToTabSelect) {
+          return;
+        }
+
+        let browser = gBrowser.selectedBrowser;
+        let cursor = browser.parentNode.querySelector("#remote-cursor");
+
+        if (!cursor) {
+          cursor = document.createElement("image");
+          cursor.setAttribute("id", "remote-cursor");
+          cursor.style.pointerEvents = "none";
+        }
+
+        // Calculate the final position based on browser's size.
+        let position = {
+          left: cursorData.ratioX * browser.boxObject.width,
+          top: cursorData.ratioY * browser.boxObject.height
+        };
+
+        // Update the cursor's position.
+        cursor.setAttribute("left", position.left);
+        cursor.setAttribute("top", position.top);
+
+        // browser's parent is a xul:stack, so positioning with left/top works.
+        browser.parentNode.appendChild(cursor);
+      },
+
+      /**
+       *  Removes the remote cursor from the screen
+       *
+       *  @param browser OPT browser where the cursor should be removed from.
+       */
+      removeRemoteCursor: function(browser) {
+        browser = browser || gBrowser.selectedBrowser;
+        let cursor = browser.parentNode.querySelector("#remote-cursor");
```

Comment:
There should only be one element with that `id`, so there's no need to reference `browser`/`gBrowser`. Just `getElementById` and remove it if it exists.